gwpy/timeseries/tests/test_timeseries.py

```diff
@@ -842,6 +842,53 @@ def test_demodulate(self):
         assert ph.unit == 'rad'
         utils.assert_allclose(ph.value, phase, rtol=1e-5)
 
+    def test_heterodyne(self):
```

Comment:
This looks good, though if you use my other suggestion, the unit test can be cleaned up and simplified.

Reply:
I'll implement your suggestion then clean up the test appropriately.